src/main/objects/HarmonicFunction.js

```diff
@@ -1,34 +1,89 @@
-function HarmonicFunction(functionName, degree, position, revolution, delay, extra, omit, down, system, mode) {
-    this.functionName = functionName // "T", "S", "D"
-    this.degree = degree             // int 
-    this.position = position         // should be string f.e. "7<" 
-    this.revolution = revolution     // should be string f.e. "7>>"
-    this.delay = delay               //delayed components list 
-    this.extra = extra               //extra components list [] | ["1", "3"]
-    this.omit = omit                 //omitted components list [] | ["7>", "9"]
-    this.down = down                 //true or false
-    this.system = system             // "open" | "close"
-    this.mode = mode                // "major" | "minor" | undefined
-
-    this.getSymbol = function () {
-        return this.down ? (this.functionName + "down" + this.extra) : (this.functionName + this.extra)
+.import "./Scale.js" as Scale
+.import "./ChordComponentManager.js" as ChordComponentManager
+.import "./Utils.js" as Utils
+.import "./HarmonicFunctionValidator.js" as HarmonicFunctionValidator
+
+var DEBUG = false;
+
+function HarmonicFunction2(params){
+    // Properties:
+    // functionName          "T", "S", "D"
+    // degree               int
+    // position             should be string f.e. "<7" or "7>" or just "7"
+    // revolution           should be string f.e. "<6" or "6>" or just "6"
+    // delay                delayed components list
+    // extra                extra components list [] | ["9", "7>"]
+    // omit                 omitted components list [] | ["1", "5"]
+    // down                 true or false
+    // system               "open" | "close" | undefined
+    // mode                 "major" | "minor"
+
+    var cm = new ChordComponentManager.ChordComponentManager();
+
+    this.functionName = params["functionName"];
+    this.degree = params["degree"] === undefined ? getDegreeFromFuntionName(this.functionName) : params["degree"];
+    this.position = params["position"];
+    this.revolution = params["revolution"] === undefined ? (params["down"] === true ? "1>" : "1") : params["revolution"];
```

Comment:
Tu możnaby zastosować tę logikę, o której mówiłem. Czyli jeśli undefined, to sprawdzić czy mamy omit 1, później czy 3 jest w sopranie lub też jest omit to wtedy sprawdzamy 5 itd.

Reply:
Jasne, zrobię tak.